Applications/web_portal/web_app/library/source/pages/dashboard_page/dashboard_page.tsx

```diff
@@ -14,20 +16,98 @@ interface Properties {
   onLogout?: () => void;
 }
 
-interface State {}
+interface State {
+  isSideMenuOpen: boolean;
+}
 
 /** Displays the main dashboard. */
 export class DashboardPage extends React.Component<Properties, State> {
+  constructor(props: Properties) {
+    super(props);
+    this.state = {
+      isSideMenuOpen: false
+    };
+    this.toggleSideMenuIsOpen = this.toggleSideMenuIsOpen.bind(this);
+  }
   public render(): JSX.Element {
+    const getFadeTransition = (status: string) => {
+      switch(status) {
+        case 'entering':
+          return DashboardPage.FADE_TRANSITION_STYLE.entering;
+        case 'entered':
+          return DashboardPage.FADE_TRANSITION_STYLE.entered;
+        case 'exiting':
+          return  DashboardPage.FADE_TRANSITION_STYLE.exiting;
+        case 'exited':
+          return  DashboardPage.FADE_TRANSITION_STYLE.exited;
+      }
+    };
+    interface DashboardBurgerButton {
```

Comment:
Pull this interface and the DashboardBurgerButton out of the render method, make them their own standalone classes.